event-svc/src/store/sql/access/event.rs

```diff
@@ -370,17 +371,30 @@ impl CeramicOneEvent {
             all_blocks.iter().map(|row| row.block.clone()).collect(),
         )
         .await?;
+
+        // We need to match up the delivered index with each event. However all_blocks contains an
```

Comment:
We are keeping multiple copies of data in memory, can we move this logic in the SQL query itself? 
``` 
    let max_highwater = sqlx::query_as::<_, DeliveredEventBlockRow>(
        r#"
        SELECT
            event_id,
            MAX(delivered) AS max_delivered,
            block_data
        FROM
            events_table
        WHERE
            delivered > ?  -- Only fetch events with a delivered value greater than the highwater mark
        GROUP BY
            event_id
        ORDER BY
            max_delivered ASC
        LIMIT ?
        "#,
    ) 
    ```
    What do you think? Is this a better approach?

Reply:
Then we can call, 
```
.bind(highwater)
.bind(limit)
.fetch_all(pool.reader())
.await?
.into_iter()
.map(|row| {
        // Decode the CAR file from the block data
     ....

        // Add the parsed event to the list
     ....

      // Update the max highwater mark
    })
    .max()
    .unwrap_or(highwater); // Use the input highwater if no new events are found ```

Reply:
The challenge is that we need events, but we have to construct the events from the list of blocks that make up each event. So the `all_block` query needs to join with the blocks tables in order to get all the data. This PR doesn't change how much memory is needed (we were previously keeping a copy of the data in memory). I think optimizing this query should be future work.(If we find its a cause of slow down)